backend/corpora/dataset_processing/common.py

```diff
@@ -0,0 +1,114 @@
+    processing_status_type: str,
+):
+    update_db(
+        dataset_id,
+        processing_status={processing_status_type: ConversionStatus.UPLOADING},
+    )
+    logger.info(f"Uploading [{dataset_id}/{file_name}] to S3 bucket: [{artifact_bucket}].")
+    try:
+        s3_uri = DatasetAsset.upload(file_name, bucket_prefix, artifact_bucket)
+        with db_session_manager() as session:
+            logger.info(f"Updating database with  {artifact_type}.")
+            DatasetAsset.create(
+                session,
+                dataset_id=dataset_id,
+                filename=file_name,
+                filetype=artifact_type,
+                user_submitted=True,
+                s3_uri=s3_uri,
+            )
+        update_db(
+            dataset_id,
+            processing_status={processing_status_type: ConversionStatus.UPLOADED},
+        )
+
+    except Exception as e:
+        logger.error(e)
+        e.args = {processing_status_type: ConversionStatus.FAILED}
+        raise e
+
+
+def update_db(dataset_id, metadata: dict = None, processing_status: dict = None):
+    with db_session_manager() as session:
+        dataset = Dataset.get(session, dataset_id, include_tombstones=True)
+        if dataset.tombstone:
+            raise ProcessingCancelled
+
+        if metadata:
+            logger.debug("Updating metadata.")
+            dataset.update(**metadata)
+
+        if processing_status:
+            logger.debug(f"updating processing_status.{processing_status}")
+            processing_status_updater(session, dataset.processing_status.id, processing_status)
+
+
+def download_from_s3(bucket_name: str, object_key: str, local_filename: str):
+    logger.info(f"Downloading file {local_filename} from bucket {bucket_name} with object key {object_key}")
+    buckets.portal_client.download_file(bucket_name, object_key, local_filename)
+
+
+def convert_file(
+    converter: typing.Callable,
+    local_filename: str,
+    error_message: str,
+    dataset_id: str,
+    processing_status_type: str,
+) -> str:
+    logger.info(f"Converting {local_filename}")
+    start = datetime.now()
+    try:
+        update_db(
+            dataset_id,
+            processing_status={processing_status_type: ConversionStatus.CONVERTING},
+        )
+        file_dir = converter(local_filename)
+        update_db(
+            dataset_id,
+            processing_status={processing_status_type: ConversionStatus.CONVERTED},
+        )
+        logger.info(f"Finished converting {converter} in {datetime.now()- start}")
+    except Exception as e:
+        logger.exception(f"{error_message}: {e}")
+        status = {processing_status_type: ConversionStatus.FAILED}
+        raise ConversionFailed(status)
+    return file_dir
+
+
+def get_bucket_prefix(identifier):
+    remote_dev_prefix = os.environ.get("REMOTE_DEV_PREFIX", "")
+    if remote_dev_prefix:
+        return join(remote_dev_prefix, identifier).strip("/")
+    else:
+        return identifier
```

Comment:
To my eye these have all been moved from `process.py`. Are there any edits/tweaks that were made? Assuming not

Reply:
as with `logger.py` below